src/State/DelayScheduler.lua

```diff
@@ -0,0 +1,73 @@
+--!strict
+
+--[[
+  Manages batch updating of delay objects.
+]]
+
+local RunService = game:GetService("RunService")
+
+local Package = script.Parent.Parent
+local Types = require(Package.Types)
+local updateAll = require(Package.Dependencies.updateAll)
+
+type Array<T> = {[number]: T}
+type Delay = Types.Delay<any>
+type DelayData = {
+  delay: Delay,
+  start: number,
+  duration: number,
+  goal: any
+}
+
+local DelayScheduler = {}
+
+local WEAK_KEYS_METATABLE = {__mode = "k"}
+
+-- all delays to be updated
+local allDelays: Array<DelayData> = {}
+setmetatable(allDelays, WEAK_KEYS_METATABLE)
+
+--[[
+  Adds a Delay to be updated every render step.
+]]
+function DelayScheduler.add(delay: Delay)
+  table.insert(allDelays, {
+    delay = delay,
+    start = os.clock(),
+    duration = if delay._durationIsState then delay._duration:get(false) else delay._duration,
```

Comment:
This means that if the State duration changes after a change in the follower State, the original delay duration is used.
Is this the right behaviour? It might be, but I think it requires consideration.

Reply:
This was brought up in #74 and the functionality of overriding existing durations of delays in the scheduler was thought to be more like a debounce than a delay. Therefore it makes more sense to use the original duration. However, I did also look into overriding pending delays and the code implementation is simple. The question is how it should actually function.

I currently think something like this would suffice.
```lua
local state = Value(0)
local delay = Delay(state, 2, true) -- A third optional argument that will toggle on override functionality
```
So should the third argument be a constant or also accept state? I think a constant makes more sense and helps prevent edge cases but I'd like further input before implementing this.